src/resources/UsageAnalytics/Read/DataHealth/DataHealthInterfaces.ts

```diff
@@ -310,3 +310,80 @@ export interface DataHealthGetTrackingIdsResponse {
      */
     values: DataHealthFacetValue[];
 }
+
+export interface DataHealthGetEventProblemsParams extends OrganizationParamParts, TimeRangeParamParts, Paginated {
+    /**
+     * A set of event types for which the events will be returned.
+     * If not specified, events are not filtered on event types.
+     */
+    eventType?: string[];
+    /**
+     * A set of search hubs for which the events will be returned.
+     * If not specified, events are not filtered on search hub.
+     */
+    searchHub?: string[];
+    /**
+     * The tracking id to filter by when getting groups listing.
+     * If not specified, events are not filtered on tracking id.
+     */
+    trackingId?: string[];
+}
+export interface EventProblemsResponse extends PaginatedResponse {
+    /**
+     * List of the event problems.
+     */
+    problems: EventProblemsResponseItem[];
+}
+
+export interface EventProblemsResponseItem {
+    /**
+     * A set of event types for which the events will be returned.
+     * If not specified, events are not filtered on event types.
+     */
+    eventType: string;
+    /**
+     * The tracking id to filter by when getting groups listing.
+     * If not specified, events are not filtered on tracking id.
+     */
+    trackingId: string;
+    /**
+     * A set of search hubs for which the events will be returned.
+     * If not specified, events are not filtered on search hub.
+     */
+    searchHub: string;
+    /**
+     * Number of failed events.
+     */
+    failedEvents: number;
+    /**
+     * Number of total events.
+     */
+    totalEvents: number;
+    /**
+     * List of the validation errors.
+     */
+    validationErrors: ValidationErrors[];
+    /**
+     * Events sampled from the problematic events.
+     */
+    sampleEvents: string[];
+    /**
+     * Timestamps from the sample events.
+     */
+    sampleEventsTimestamps: string[];
+}
+
+interface ValidationErrors {
+    /**
+     * Validation error code.
+     */
+    code: string;
```

Comment:
Should we use a more precise type here also?

Reply:
I'm not sure what you mean here? :sweat_smile: The API returns a string, I think the safest open is to put a string here too no?

Reply:
I was wondering if there was a specific list of strings that could be extracted into a type. Intuitively, I would be inclined to think that there's a limited number of well-defined possible error codes.

However this might be an incorrect assumption. If so, then string would be fine as a type.

Reply:
Yea there's many different error code, but even if the list was pretty short, I think having all the code define here is a bad idea. IMO it's better to have just one source of truth. If we were to define them here, we would need to change the enumeration here every time we add, remove or change a error code. I think a string is the safest option.